### ISSUE TITLE
Svg not rendering when applying from a virtual dom

### ISSUE THREAD

**Issue by f95johansson** [fiduswriter/diffDOM#101]
Svgs are not being rendered correctly (not at all) when applying a virtual dom to a dom element (tested on Chrome (Mac), Safari (Mac) and Chrome (Android). The svg is seemingly inserted correctly in the html dom, but because of a bug, createElement is used instead of createElementNS and the browser wont render the svg.

The bug is that the nodeName attribute of variable objNode, in file `fromVirtual.js`,  is uppercase, but the if statement for svgs checks for lowercase (`objNode.nodeName === 'svg'`).

This issue can be fixed with two steps, check for both 'svg' and 'SVG' to run createElementNS, and then make sure the node name is always lowercase when running createElementNS. The last part is because createElementNS is case sensitive, unlike createElement which is case insensitive. And 'SVG' (and 'PATH', etc) are not valid svg elements and must be in lowercase.
That means the current code in `fromVirtual.js`:

```js
if (objNode.nodeName === 'svg' || insideSvg) {
    node = options.document.createElementNS('http://www.w3.org/2000/svg', objNode.nodeName)
    insideSvg = true
}
```

should be changed to:

```js
if (objNode.nodeName.toLowerCase() === 'svg' || insideSvg) {
    node = options.document.createElementNS('http://www.w3.org/2000/svg', objNode.nodeName.toLowerCase())
    insideSvg = true
}
```

This fixes the problem, at least from what I can test on the browsers mentioned before. I have made a pull request which makes these changes if any maintainer would like to merge it.

**Comment by rposborne** [comment]
👋  Hey friends. 

I am under the impression that this change is breaking SVG's that contain case-specific elements.  Such as https://developer.mozilla.org/en-US/docs/Web/SVG/Element/feGaussianBlur

I think the root cause needs to be addressed over in fromVirtual.  

We heavily use this library in complex SVGS and have had no issues with rendering svgs until 4.2.0 (we were on 4.1.6)

**Comment by johanneswilm** [comment]
@rposborne That is true. Could you fix it in a way that is compatible with @f95johansson previous issue? I think his concern was only with the svg tag itself and we probably shouldn't need to change the case of anything to make this work.

**Comment by rposborne** [comment]
@f95johansson Will you provide a example / failure case that I can test against?  All you reasoning appears sound, I just don't know how you found yourself with a tag with `SVG` vs `svg` I wonder if this is an issue with the svgs being used.

**Comment by johanneswilm** [comment]
If the issue is just with the `SVG` tag itself, and not anything inside of the tag, would something like this work for both of you:

```js
if (objNode.nodeName.toLowerCase() === 'svg' || insideSvg) {
    node = options.document.createElementNS('http://www.w3.org/2000/svg', 'svg')
    insideSvg = true
} else if (insideSvg) {
    node = options.document.createElementNS('http://www.w3.org/2000/svg', objNode.nodeName)
}
```

maybe?

**Comment by johanneswilm** [comment]
Where exactly does the 'SVG' with capital letters get introduced here?

**Comment by johanneswilm** [comment]
@rposborne have tried to contact @f95johansson . If we don't hear from him within another two days, I think we should just go ahead and revert the commit and make a patch release with that.

**Comment by johanneswilm** [comment]
@rposborne: @f95johansson contacted me and agreed to revert the change. After looking at the issue more closely, I believe the issue is that the SVG tag itself can be in any case in the original HTML (for example "SvG"), but that always creates a lowercase "svg" node by the browser. All the nodes inside of the SVG node are however case sensitive, so they should not be changed. I have now changed the code accordingly and published a version 4.2.1. Please let me know if either of you have issues with this and we will try out something else.

**Comment by rposborne** [comment]
Thanks so much @johanneswilm and @f95johansson. Sorry to be a pain!
